quickstart/101-key-vault-key/main.tf

```diff
@@ -0,0 +1,62 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "random_string" "azurerm_key_vault_name" {
+  length  = 13
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_key_vault" "vault" {
+  name                       = (var.vault_name != "") ? var.vault_name : "vault-${random_string.azurerm_key_vault_name.result}"
+  location                   = azurerm_resource_group.rg.location
+  resource_group_name        = azurerm_resource_group.rg.name
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  sku_name                   = var.sku_name
+  soft_delete_retention_days = 7
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
```

Comment:
We'll have a testing issue here. The `data.azurerm_client_config.current` only works for service principal, but in our e2e test environment we're using managed identity, the `data.azurerm_client_config.current.object_id` could be empty string in e2e test.

Now in our e2e test we'll setup an environment variable named [`TF_VAR_msi_id`](https://github.com/Azure/terraform/blob/master/test/e2e/quickstart_test.go#L14-L18), that means if you've added a variable named `msi_id` like this way, during the test it would be assigned the identity id that we're using for the test:

```hcl
variable "msi_id" {
  type     = string
  default = null
}
```

I would recommend a `local` expression like this:

```hcl
locals {
  current_user_id = coalesce(var.msi_id, data.azurerm_client_config.current.object_id)
}
```

Then change line 30 to:

```hcl
object_id = local.current_user_id
```

Reply:
@lonegunmanb @grayzu Please also check my description for this var. Thanks!